### PR TITLE
feat(meshmetric): manually remove regex

### PR DESCRIPTION
**Pull request by slonka**
closes https://github.com/kumahq/kuma/issues/9774

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
Don't we need to follow some deprecation time here?

**Comment by slonka** [comment]
> Don't we need to follow some deprecation time here?

MeshMetric was in alpha and noone is using it on our hosted platform. And even if someone was using it on-prem the only bad thing that could happen is that their metric is not filtered.

**Comment by slonka** [comment]
> Don't we need to follow some deprecation time here?

image

Hmm... actually only we marked OTEL as `alpha` not the whole policy :|

**Comment by lahabana** [comment]
Ha this is part of MeshMetric. I think it's fine then I thought it was a more global config

File: app/kuma-dp/pkg/dataplane/metrics/server.go [hunk 1]
```diff
@@ -74,7 +74,6 @@ func AddSidecarParameters(sidecar *v1alpha12.Sidecar) func(queryParameters url.V
 	values := v1alpha1.EnvoyMetricsFilter(sidecar)
 
 	return func(queryParameters url.Values) url.Values {
-		queryParameters.Set("filter", values.Get("filter"))
 		queryParameters.Set("usedonly", values.Get("usedonly"))
```
Comment (Automaat) [inline app/kuma-dp/pkg/dataplane/metrics/server.go:77]: is the usedonly still relevant here? don't we set this on otel producer?

**Review by Automaat — APPROVED** [review]

**Pull request merged**